tests/roundtrip.rs

```diff
@@ -0,0 +1,144 @@
+extern crate tempfile;
+extern crate escargot;
+extern crate reqwest;
+extern crate unzip;
+
+use tempfile::tempdir;
+use std::fs::File;
+//use std::path::Path;
+use std::io::{self, Write};
+use std::vec::Vec;
+//use std::env;
+
+#[macro_use]
+extern crate duct;
+
+//use duct::cmd;
+
+#[test]
+fn roundtrip() { //return -> std::io::Result<()> ?
+//TODO: wrap test tasks in a function that returns Result so we can safely
+//dispose of temp directories, and kill the server if test fails?
```

Comment:
Very good point. There is a more advanced but very cool technique we can use here: Implementing [`Drop`](https://doc.rust-lang.org/1.30.0/std/ops/trait.Drop.html) for our structures. When a variable goes out of scope (this includes when the function exits due to a panic), its `Drop::drop` implementation will be called.

This is often documented on "interesting" structures. The docs of [`tempfile::TempDir`](https://docs.rs/tempfile/3.0.4/tempfile/struct.TempDir.html) mentions:

> Once the TempDir value is dropped, the directory at the path will be deleted, along with any files and directories it contains.

This is nice, and often enough. [`std::process::Child`](https://doc.rust-lang.org/std/process/struct.Child.html) says this, however:

> There is no implementation of Drop for child processes, so if you do not ensure the Child has exited then it will continue to run, even after the Child handle to the child process has gone out of scope.

So, we'll have to deal with that ourselves. How? Let's put that running server process (or a handle to it, at least), in a new data structure. (This is a very Rust-y thing to do.) For example, here is a draft of a structure that holds on to the [`std::process::Child`](https://doc.rust-lang.org/std/process/struct.Child.html) we get from [`Command::spawn()`](https://doc.rust-lang.org/std/process/struct.Command.html#method.spawn) (this assumes we use `std::process`, which I think suffices here, and is what we get from using escagot's `run`; you can also use `duct::Handle`):

```rust
struct Server { process: ::std::process::Child };

impl Server {
    fn start(command: ::std::process::Command) -> Result<Server, ::std::io::Error> {
        let process = command.stdout(Stdio::null()).spawn()?;
        Ok(Server { process })
    }
}

impl Drop for Server {
    fn drop(&mut self) {
        let _ = self.process.kill(); // ignore any error here -- your choice, really
    }
}
```

Now, if you have a `let server = Server::start(escargot::CargoBuild::new().run().command())?;` somewhere, it'll automatically call `kill` when it goes out of scope.

This is a very neat pattern -- and defining structures that own/deal with resources is what Rust is all about, after all ;)

You can read more about Drop here: https://doc.rust-lang.org/1.30.0/book/2018-edition/ch20-03-graceful-shutdown-and-cleanup.html